terra/compute/base.py

```diff
@@ -186,5 +191,91 @@ def configuration_map_service(self, service_info):
 
     return service_info.volumes
 
+  @staticmethod
+  def configure_logger(sender, **kwargs):
+    if settings.terra.zone == 'controller':
+      print("SGR - setting up controller logging")
+
+      # Setup log file for use in configure
+      sender._log_file = os.path.join(settings.processing_dir,
+                                      terra.logger._logs.default_log_prefix)
+      os.makedirs(settings.processing_dir, exist_ok=True)
+      sender._log_file = open(sender._log_file, 'a')
+      sender.main_log_handler = StreamHandler(stream=sender._log_file)
+      sender.root_logger.addHandler(sender.main_log_handler)
+
+
+      # setup the TCP socket listener
+      sender.tcp_logging_server = LogRecordSocketReceiver(
+          settings.logging.server.hostname, settings.logging.server.port)
+      print('SGR - About to start TCP server...')
+
+      listener_thread = threading.Thread(
+          target=sender.tcp_logging_server.serve_until_stopped)
+      listener_thread.setDaemon(True)
+      listener_thread.start()
+
+      # Wait up to a second, to make sure the thread started
+      for _ in range(1000):
+        if sender.tcp_logging_server.ready:
+          break
+        time.sleep(0.001)
+
+      # Auto cleanup
+      @atexit.register
+      def cleanup_thread():
+        print("SGR - Sending cease and desist to LogRecordSocketReceiver")
+        sender.tcp_logging_server.abort = 1
+        listener_thread.join(timeout=5)
+        if listener_thread.is_alive():
+          print("SGR - LogRecordSocketReceiver thread did not die")
+        print("SGR - LogRecordSocketReceiver died!")
+    elif settings.terra.zone == 'runner':
+      print(f"SGR - setting up BaseCompute:runner logging")
+
+      sender.main_log_handler = SocketHandler(
+          settings.logging.server.hostname, settings.logging.server.port)
+      sender.root_logger.addHandler(sender.main_log_handler)
+
+      # Now in configure_logger, you are able to access settings and determine
+      # whether there should be a stderr handler or not. If you don't so this,
+      # both the master controller and service runner will output the same log
+      # messages, duplicating output on stderr.
+      sender.root_logger.removeHandler(sender.stderr_handler)
```

Comment:
You mention a variable in settings to determine whether there should be a stderr handler, but you don't seem to be checking it.

Also, i think the comment should read, "will output the same log messages, duplicating and potentially interleaving output (as stdout/err are not threadsafe) on stderr"

Reply:
I don't think this variable ever came to fruition, and is no longer needed or desired, or am I missing something that was done?

Reply:
It's not my comment. If it's no longer needed just delete the comment